ethereumj-core/src/main/java/org/ethereum/sync/SyncManager.java

```diff
@@ -91,9 +104,8 @@ public void accept(BlockWrapper blockWrapper) {
     private CountDownLatch receivedHeadersLatch = new CountDownLatch(0);
     private CountDownLatch receivedBlocksLatch = new CountDownLatch(0);
 
-    private Thread syncQueueThread;
-    private Thread getHeadersThread;
-    private Thread getBodiesThread;
+    // All workers-like thread loops to simplify shutdown
+    private List<Thread> threadLoops = new ArrayList<>();
```

Comment:
I would also return back these fields to simplify further class reading. (each of these threads has its own essential destiny :) )